applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -21,91 +21,101 @@
             auto result = std::make_unique<GeoMechanicsNewtonRaphsonStrategy<TSparseSpace,
                     TDenseSpace,
                     TLinearSolver>>(rModelPart,
-                    scheme,
-                    solver,
-                    criteria,
-                    builder_and_solver,
-                    strategy_parameters,
-                    max_iterations,
-                    calculate_reactions,
-                    reform_dof_set_at_each_step,
-                    move_mesh_flag);
+                                    scheme,
+                                    solver,
+                                    criteria,
+                                    builder_and_solver,
+                                    strategy_parameters,
+                                    max_iterations,
+                                    calculate_reactions,
+                                    reform_dof_set_at_each_step,
+                                    move_mesh_flag);
             result->SetEchoLevel(echo_level);
             return result;
+        } else if (rSolverSettings[strategy_type].GetString() == "line_search") {
+            const std::vector<std::string> strategy_entries = {"max_iteration",
+                                                               "compute_reactions",
+                                                               "max_line_search_iterations",
+                                                               "first_alpha_value",
+                                                               "second_alpha_value",
+                                                               "min_alpha",
+                                                               "max_alpha",
+                                                               "line_search_tolerance",
+                                                               "move_mesh_flag",
+                                                               "reform_dofs_at_each_step",
+                                                               "echo_level"};
+
+            auto strategy_parameters = ExtractStrategyParameters(rSolverSettings, strategy_entries);
+            auto result = std::make_unique<LineSearchStrategy<TSparseSpace,
+                    TDenseSpace,
+                    TLinearSolver>>(rModelPart,
+                                    scheme,
+                                    solver,
+                                    criteria,
+                                    strategy_parameters);
+            return result;
         }
 
         return nullptr;
     }
 
 private:
-    static Parameters ExtractStrategyParameters(const Parameters &rSolverSettings)
-    {
+    static Parameters ExtractStrategyParameters(const Parameters &rSolverSettings,
+                                                const std::vector<std::string> strategy_entries) {
```

Comment:
This can be a const& instead of only const
```suggestion
                                                const std::vector<std::string>& strategy_entries) {
```

Reply:
done